src/soundio/soundmanagerutil.h

```diff
@@ -197,6 +197,52 @@ class AudioDestination {
 
 typedef AudioPath::AudioPathType AudioPathType;
 
+class SoundDeviceId {
+  public:
+    QString name;
+    // The "hw:X,Y" device name. Remains an empty string if not using ALSA
+    // or using a non-hw ALSA device such as "default" or "pulse".
+    QString alsaHwDevice;
+    int portAudioIndex;
+
+    QString debugName() const {
```

Comment:
Or overload `QDebug operator<<(QDebug, const DeviceId&)`?

Reply:
Ok, the overload already exists. Those multiple representations are slightly confusing.

Reply:
I tried defining only the operator<<, but there are use cases for this debugName in the name for Tracers in SoundDevicePortAudio. Do you have suggestions for another way to handle this?